pkg/apis/autoscaling/v1alpha1/pa_types.go

```diff
@@ -155,15 +164,56 @@ func (pa *PodAutoscaler) ScaleBounds() (min, max int32) {
 	return
 }
 
-func (pa *PodAutoscaler) MetricTarget() (target int32, ok bool) {
+// Target returns the target annotation value or false if not present.
+func (pa *PodAutoscaler) Target() (target int32, ok bool) {
 	if s, ok := pa.Annotations[autoscaling.TargetAnnotationKey]; ok {
 		if i, err := strconv.Atoi(s); err == nil {
+			if i < 1 {
+				return 0, false
+			}
 			return int32(i), true
 		}
 	}
 	return 0, false
 }
 
+// Window returns the window annotation value or false if not present.
+func (pa *PodAutoscaler) Window() (window time.Duration, ok bool) {
+	if s, ok := pa.Annotations[autoscaling.WindowAnnotationKey]; ok {
+		d, err := time.ParseDuration(s)
+		if err != nil {
+			return 0, false
+		}
+		if d < time.Second*6 {
+			return 0, false
+		}
+		return d, true
+	}
+	return 0, false
+}
+
+// WindowPanicPercentage returns panic window annotation value or false if not present.
+func (pa *PodAutoscaler) WindowPanicPercentage() (percentage float64, ok bool) {
+	percentage, ok = pa.annotationFloat64(autoscaling.WindowPanicPercentageAnnotationKey)
+	// Panic window cannot be bigger than stable window. Or vanishingly small
+	// (cannot divide by zero).
+	if !ok || percentage > 100.0 || percentage < 1.0 {
```

Comment:
<1.0 is not vanishingly small, e.g. (0.95). So we might want to be more precise with the comment.

Reply:
Agree.  I'll provide a little detail about the subtleties here and what values won't make sense.

Reply:
Could you provide detail about the requirement for other labels too? E.g. TargetPanicPercentage should be bigger than 110.

Reply:
I've moved all the min/max values to constants in `register.go` and added more comprehensive comments.